### ISSUE TITLE
Compatibility with web essentials 2015 broken

### ISSUE THREAD

**Issue by valeriob** [madskristensen/TypeScriptDefinitionGenerator#13]
The Typescript definition code generation in web essential used to generate a file with this rule :
ViewModels.cs =>ViewModels.cs.d.ts we have a team that is gradually migrating to vs 2017 and the only issue we have is that this extension does not work with definitions generated with vs 2015 because it uses a different convention ViewModels.cs =>ViewModels.d.ts.
This little thing causes pain :D

**Comment by styfle** [comment]
I would submit a pull request for this but I have no experience writing extensions.

I found [line 68](https://github.com/madskristensen/TypeScriptDefinitionGenerator/blob/11f43c835f68c8a45a026df40c2b7419f76e825f/src/Commands/ToggleCustomTool.cs#L68) and [line 85](https://github.com/madskristensen/TypeScriptDefinitionGenerator/blob/11f43c835f68c8a45a026df40c2b7419f76e825f/src/Commands/ToggleCustomTool.cs#L85) is likely what needs to be changed.

```diff
- string dtsFile = Path.ChangeExtension(_item.FileNames[1], Constants.FileExtension);
+ string dtsFile = _item.FileNames[1] + Constants.FileExtension;
```

@madskristensen What do you think?

**Comment by raboud** [comment]
Seems like this is the root of several issues.

Fixing would be easy but need direction whether we want to leave in the original file extension of not.

A single static method to Generate the file name called everywhere including HasIntellisense

**Comment by valeriob** [comment]
Given #18 how does it change the status of this issue ?

**Comment by raboud** [comment]
I can think of 2 possible ways to address this.

1) mod the routine to just do it the old way.
2) add a configuration option to generate it the old way or the new way

I can do either just really need Mad's input how he would like it fixed. I have fixes for several other things but haven't done the pull request for them yet.

**Comment by valeriob** [comment]
Great @raboud !
I really hope your work will be merged soon, do you have a fix for #19 ? We are starting to use vs2017, and it's very anoying ! Given that in vs 2015 with web essential it worked great :(

**Comment by styfle** [comment]
@raboud I think a maintaining compatibility with WebEssentials 2015 is extremely important. This should be the default behavior if you allow options to change.

If this extension doesn't maintain that old behavior, then we might as well switch to [TypeWriter](https://github.com/frhagn/Typewriter) because it can be configured far better than this extension.

**Comment by raboud** [comment]
Please Try Build 1.1.24 to see if your issues are corrected.

**Comment by styfle** [comment]
So far this looks great! I'll have to try it for a couple weeks to know for sure and I'll let you know if I find any quirks!

**Comment by nhammadi** [comment]
@madskristensen issue resolved with 1.1.24 release, you should close this issue.
